### PR TITLE
Docker build enabled for Mac. Rviz2 is not support through docker. Ex…

### PR DESCRIPTION
**Pull request by rzjin**
…ploring Rocker for GUI support

### CODE REVIEW

**Comment by rzjin** [comment]
Debugging logs:
https://childlike-base-97e.notion.site/Turtlebot-Setup-for-MacOS-91d3b36bf4584d72b1d4f78b6e5666f0

**Comment by proan** [comment]
Does this work on Apple Silicon Macs or Intel or both?

I don't have a Mac to test on, but I'll make sure the changes don't impact my Ubuntu machines.

File: turtlebot2_bringup/CMakeLists.txt [hunk 1]
```diff
@@ -17,6 +17,15 @@ install(
...
 )
 
+# assume built-in pthreads on MacOS
+if(APPLE)
+    set(CMAKE_THREAD_LIBS_INIT "-lpthread")
+    set(CMAKE_HAVE_THREADS_LIBRARY 1)
+    set(CMAKE_USE_WIN32_THREADS_INIT 0)
+    set(CMAKE_USE_PTHREADS_INIT 1)
+    set(THREADS_PREFER_PTHREAD_FLAG ON)
+endif()
```
Comment (proan) [inline turtlebot2_bringup/CMakeLists.txt]: Move this block up to line 7. This way it's separated from the ROS parts.

**Comment by proan** [comment]
Building sophus from source takes a lot of memory, so I also have to reduce parallel workers to 1. I think there may be a version pre-build for Humble and Iron now, but I haven't been able to put it all together yet.

**Comment by rzjin** [comment]
> Does this work on Apple Silicon Macs or Intel or both?
> 
> I don't have a Mac to test on, but I'll make sure the changes don't impact my Ubuntu machines.

I believe this should be specific to ARM based Macs but I don't have an intel based Mac to validate this assumption

**Comment by proan** [comment]
Works on my robot, so let's merge. We can adjust when we have a better suite of test environments.

**Pull request merged**